components/contact-list.tsx

```diff
@@ -23,90 +26,115 @@ export function ContactList(props: ContactListProps) {
     a: anchorTransformer,
   };
 
+  const parentRef = useRef<HTMLUListElement>(null);
+
+  const rowVirtualizer = useVirtual({
+    size: props.data.length,
+    parentRef,
+  });
+
   return (
-    <div className="bg-white shadow overflow-hidden sm:rounded-md">
-      <ul className="divide-y divide-gray-200">
-        {props.data.map((contact, index) => (
-          <li key={index}>
-            <div className="px-4 py-4 sm:px-6 relative hover:bg-gray-50">
-              <div className="flex items-center justify-between">
-                <Link href={`/provinces/${props.provinceSlug}/${contact.slug}`}>
-                  <a className="text-sm font-semibold text-blue-600 truncate block helper-link-cover">
-                    {isNotEmpty(contact.penyedia)
-                      ? contact.penyedia
-                      : contact.keterangan}
-                  </a>
-                </Link>
-                <div className="ml-2 flex-shrink-0 flex">
-                  <p className="px-2 inline-flex text-xs leading-5 font-semibold rounded-full bg-yellow-100 text-yellow-800">
-                    {contact.kebutuhan}
+    <div className="bg-white shadow sm:rounded-md overflow-hidden">
+      <ul
+        className="divide-y divide-gray-200 h-screen relative overflow-y-auto overflow-x-hidden"
+        ref={parentRef}
+      >
+        {rowVirtualizer.virtualItems.map((virtualRow) => {
+          const contact: Contact = props.data[virtualRow.index];
+          return (
+            <li
+              key={virtualRow.index}
+              ref={virtualRow.measureRef}
+              style={{
+                position: "absolute",
+                top: 0,
+                left: 0,
+                width: "100%",
```

Comment:
Yes, I've tried that, Mas, but unfortunately, it does not solve the main problem. Instead, I use `totalSize` to create an inner container of the list.

My actual concern is that we ended up having two vertical scrolls which I think is a bad scrolling experience, particularly on a mobile viewport. Are you okay with that behaviour, Mas? Would you mind confirming that scrolling experience yourself via the [deploy preview](https://deploy-preview-134--wargabantuwarga.netlify.app/provinces/jawa-timur-15), Mas? :sungkem:

https://user-images.githubusercontent.com/24476578/126411810-1e961ec3-bac2-4c5b-9db1-3958a3313e27.mp4

https://user-images.githubusercontent.com/24476578/126412089-67d44f83-b8d4-4f5b-9a48-c72a9dee2028.mp4

Reply:
Yes, I am aware of it. In my opinion, before we need to merge this, we need to simplify the Search and Filter UI, because it takes too much space.

I think we can use something like this to squeeze the filtering & sorting UI into a smaller space, what do you think, @resir014?

https://user-images.githubusercontent.com/6315466/126420178-2ed315fc-ee12-476d-a77f-621ee8c6685b.MOV

Reply:
Yes, agree @zainfathoni.

I'm not sure about the double scroll though. It feels a little too janky. Unless we can simplify the filter we shouldn't use it.

Reply:
@zainfathoni Let's create a separate issue for the simplified filter. I can work on this later tonight.